express-mock-af/routes/reporting-ui/sse.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express');
+const Utils = require('../../utils/Utils');
+const router = express.Router();
+
+/**
+ * This endpoint functions as an event source that sends a message every time a new file is written.
+ * This is base on {@link https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events SSE } (Server-Sent Events)
+ */
+router.get('/reload', (req, res) => {
+    res.writeHead(200, {
+        'Access-Control-Allow-Origin': '*',
+        'Content-Type': 'text/event-stream; charset=utf-8',
+        'Cache-Control': 'no-cache',
+        'Connection': 'keep-alive',
+        'Content-Encoding': 'none'
+    });
+
+    const subscription = Utils.fileWritten$.subscribe({
+        next: async ({ content, topic}) => {
+            res.write(`event: ${topic}\ndata: A new file has been written\n\n`);
```

Comment:
Do we really need a topic, i thought everytime it writes a consumption report it also writes a metrics report and vice versa?

Reply:
No it doesnt f.e. topic 6 contains metrics reports but not consumption ones => we need to distiguish beetween write events.

![Screenshot 2024-06-19 at 09 20 08](https://github.com/KonsumGandalf/awt-pj-ss24-5gmag-2/assets/66205241/3d963249-3fd3-47ec-86dd-cea0bba63eeb)